src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -605,6 +605,24 @@ def _service_instance_action(self, action, svc_type, svc_id):
         orchestrator.raise_if_exception(completion)
         return HandleCommandResult(stdout=completion.result_str())
 
+    @orchestrator._cli_write_command(
+        'orchestrator mgr add',
```

Comment:
From a CLI perspective, we have

```bash
ceph orchestrator rgw add <name> # adds a cluster of RGW daemons
ceph orchestrator mds add <name>  # adds a cluster of MDS daemons
ceph orchestrator nfs add <name>  # adds a cluster of NFS daemons
ceph orchestrator osd create <drive_groups>  # adds a lot of OSDs at once
ceph orchestrator mgr update <num> [host, host:network...]  # updates the deployment spec for MGRs
```
In this context
```
orchestrator mgr add <num> [host, host:network...]  # add MGRs
```
a bit wired. 

Can we find a better UX?

Reply:
In the rgw case it's realm and zone.  In the fs case it's fs-name.  I think that part is natural, but yes, it is annoying!  and for each there is update/add/rm.  And we're about to add prometheus, grafana, alertmanager, and nodeexporter to the list.  And rgw-nfs and iscsi eventually.

I wish we could do just ``ceph orchestrator add thingname [placement]``, but I'm not sure it is a good idea to include a bunch of optionals like --fs-name and --realm-name that are specific to certain types.  WDYT?

Reply:
I think we should have a conversation about how to streamline the CLI so be less fragmented.  This is similar to what the rbd-mirror add/rm are doing, though; I think it's fine to start here.

Reply:
Two sides here:

1. let's look at K8c CRDs.
2. having a symmetric CLI

Having something like this would be an idea
```
$ ceph orchestrator {host,device,service} ls
$ ceph orchestrator <service-type> {create,rm,get,replace,extend,shrink} [<service-name>]
```

Reply:
or just
```
ceph orchestrator rm <service-name>
ceph orchestrator add <service-name> [placement]
ceph orchestrator update <service-name> [placement]
```
no reason IMO to separate out the service type and name when everywhere we display them we smoosh them together with a ``.``.

Reply:
see https://pad.ceph.com/p/orchestrator_cli

Reply:
see my comments to it.